pkg/tfbridge/config_encoding.go

```diff
@@ -68,19 +68,20 @@ func (*ConfigEncoding) tryUnwrapSecret(encoded any) (any, bool) {
 }
 
 func (enc *ConfigEncoding) convertStringToPropertyValue(s string, typ shim.ValueType) (resource.PropertyValue, error) {
-	// If the schema expects a string, we can just return this as-is.
-	if typ == shim.TypeString {
-		return resource.NewStringProperty(s), nil
-	}
-
-	// Otherwise, we will attempt to deserialize the input string as JSON and convert the result into a Pulumi
+	// We attempt to deserialize the input string as JSON and convert the result into a Pulumi
 	// property. If the input string is empty, we will return an appropriate zero value.
+	//
+	// We need to do this for all incoming strings, even if the target type is also a string. This is how
+	// we account for secret or computed string values.
```

Comment:
Let's add some detail here on _how_ this accounts for secrets or computed string values.